data/src/main/java/io/gnosis/data/utils/SafeTxHash.kt

```diff
@@ -0,0 +1,108 @@
+package io.gnosis.data.utils
+
+import io.gnosis.data.models.DetailedExecutionInfo
+import io.gnosis.data.models.TransactionDetails
+import io.gnosis.data.models.TransactionInfo
+import io.gnosis.data.models.TransferInfo
+import pm.gnosis.crypto.utils.Sha3Utils
+import pm.gnosis.model.Solidity
+import pm.gnosis.utils.hexToByteArray
+import pm.gnosis.utils.toHex
+import java.math.BigInteger
+
+private const val ERC191_BYTE = "19"
+private const val ERC191_VERSION = "01"
+
+fun calculateSafeTxHash(
```

Comment:
I would love to see unit tests for this. There are some hardcoded values too, that I am guessing come from the legacy app and are just what they are. Could we port the unit tests from the legacy app over to this one?